.github/workflows/changes.yml

```diff
@@ -0,0 +1,41 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name: Change detection
+
+on:
+  workflow_call:
+    outputs:
+      kubernetes:
+        description: "whether current workflow touches deploy/kubernetes dir or not"
+        value: ${{ jobs.changes.outputs.kubernetes }}
+
+jobs:
+  changes:
+    runs-on: ubuntu-20.04
+    name: detect code change
```

Comment:
Better to find a shorter name

<img width="258" alt="image" src="https://user-images.githubusercontent.com/5821159/219652077-a0718745-aafe-478e-ba2d-0325c882b973.png">